### PR TITLE
[backport] ray: Cleanup opendatahub Namespace

### PR DESCRIPTION
**Pull request by zdtsw**
backport https://github.com/red-hat-data-services/rhods-operator/pull/246
ref :https://issues.redhat.com/browse/RHOAIENG-5514

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -688,3 +691,44 @@ func RemoveLabel(cli client.Client, objectName string, labelKey string) error {
...
+	for _, owner := range foundNamespace.OwnerReferences {
+		if owner.Kind == "DataScienceCluster" {
+			isOwnedByDSC = true
+		}
+	}
+	if !isOwnedByDSC {
+		return nil
+	}
+
+	// Check if namespace has pods running
```
Comment (ykaliuta) [inline pkg/upgrade/upgrade.go:715]: We do care about pods only, not configmaps and so on?

**Comment by ykaliuta** [comment]
I hope we can replace all that cleanup code with more generic solution when it's ready.

**Review by ykaliuta — APPROVED** [review]

File: pkg/upgrade/upgrade.go (hunk 1 above)
Comment (zdtsw) [reply pkg/upgrade/upgrade.go:715]: wont be able to know what exactly can be created in the namespace, pod is the minimum we can check to indicate if this namespace is being used or not

**Pull request merged**